internal/config/generator.go

```diff
@@ -403,65 +498,75 @@ func (c *JiraCLIConfigGenerator) configureServerAndLoginDetails() error {
 		if ans.Server != "" {
 			c.value.server = strings.TrimSpace(ans.Server)
 		}
-		if ans.Login != "" {
-			c.value.login = strings.TrimSpace(ans.Login)
-		}
-	}
 
-	return c.verifyLoginDetails(c.value.server, c.value.login)
+	}
+	return c.verifyServer()
 }
 
-func (c *JiraCLIConfigGenerator) verifyLoginDetails(server, login string) error {
-	s := cmdutil.Info("Verifying login details...")
-	defer s.Stop()
+func (c *JiraCLIConfigGenerator) setApiServer() {
+	if c.value.authType == jira.AuthTypeOAuth {
+		c.value.apiServer = fmt.Sprintf("%s/%s", apiServer, c.value.oauth.cloudId)
+	} else {
+		c.value.apiServer = c.value.server
+	}
+}
 
-	server = strings.TrimRight(server, "/")
+func (c *JiraCLIConfigGenerator) verifyServer() error {
+	c.setApiServer()
+	c.value.server = strings.TrimRight(c.value.server, "/")
+	return c.verifyLoginDetails()
+}
 
-	c.jiraClient = api.Client(jira.Config{
-		Server:   server,
-		Login:    login,
+func (c *JiraCLIConfigGenerator) generateJiraConfig() jira.Config {
+	config := jira.Config{
+		Server:   c.value.apiServer,
+		Login:    c.value.login,
 		Insecure: &c.usrCfg.Insecure,
 		AuthType: &c.value.authType,
 		Debug:    viper.GetBool("debug"),
-		MTLSConfig: jira.MTLSConfig{
+	}
+
+	switch c.value.authType {
+	case jira.AuthTypeOAuth:
+		config.APIToken = c.value.oauth.accessToken
+	case jira.AuthTypeMTLS:
+		config.MTLSConfig = jira.MTLSConfig{
 			CaCert:     c.value.mtls.caCert,
 			ClientCert: c.value.mtls.clientCert,
 			ClientKey:  c.value.mtls.clientKey,
-		},
-	})
+		}
+	}
+	return config
+}
+
+func (c *JiraCLIConfigGenerator) verifyLoginDetails() error {
+	s := cmdutil.Info("Verifying login details...")
+	defer s.Stop()
+	// Configure JIRA client based on auth type
+	config := c.generateJiraConfig()
+	c.jiraClient = api.Client(config)
+
 	ret, err := c.jiraClient.Me()
 	if err != nil {
 		return err
 	}
 	if c.value.authType == jira.AuthTypeBearer {
-		login = ret.Login
+		c.value.login = ret.Login
 	}
 
-	c.value.server = server
-	c.value.login = login
 	c.value.timezone = ret.Timezone
 
 	return nil
 }
 
-func (c *JiraCLIConfigGenerator) configureServerMeta(server, login string) error {
+func (c *JiraCLIConfigGenerator) configureServerMeta() error {
 	s := cmdutil.Info("Fetching server details...")
 	defer s.Stop()
 
-	server = strings.TrimRight(server, "/")
-
-	c.jiraClient = api.Client(jira.Config{
-		Server:   server,
-		Login:    login,
-		Insecure: &c.usrCfg.Insecure,
-		AuthType: &c.value.authType,
-		Debug:    viper.GetBool("debug"),
-		MTLSConfig: jira.MTLSConfig{
-			CaCert:     c.value.mtls.caCert,
-			ClientCert: c.value.mtls.clientCert,
-			ClientKey:  c.value.mtls.clientKey,
-		},
-	})
+	if c.jiraClient != nil {
+		config := c.generateJiraConfig()
+		c.jiraClient = api.Client(config)
+	}
 	info, err := c.jiraClient.ServerInfo()
```

Comment:
This part looks strange to me

So we had a non nil jiraClient, and we set a new one ?

This kind of code is usually when the jiraClient is nil